discovery-provider/src/utils/helpers.py

```diff
@@ -343,6 +343,33 @@ def create_track_route_id(title, handle):
 
     return f"{sanitized_handle}/{sanitized_title}"
 
+
+def create_track_slug(title, collision_id=0):
+    """ Converts the title of a track into a URL-friendly 'slug'
+
+    Strips special characters, replaces spaces with dashes, converts to
+    lowercase, and appends a collision_id if non-zero.
```

Comment:
just confirming, how does this behave w/null terminator?

Reply:
Added the null character to the list of replacements just to be doubly sure!